components/mpas-framework/src/tools/registry/gen_inc.c

```diff
@@ -1750,20 +1764,21 @@ int parse_var(FILE *fd, ezxml_t registry, ezxml_t superStruct, ezxml_t currentVa
 
 	// Determine name of pointer for this field.
 	set_pointer_name(type, ndims, pointer_name, time_levs);
-	if (time_levs > 1) {
-		fortprintf(fd, "      allocate(%s(%d))\n", pointer_name, time_levs);
-	} else {
-		fortprintf(fd, "      allocate(%s)\n", pointer_name);
-	}
 
-	for(time_lev = 1; time_lev <= time_levs; time_lev++){
-		if (time_levs > 1) {
-			snprintf(pointer_name_arr, 1024, "%s(%d)", pointer_name, time_lev);
+	if (time_levs > 1 || time_levs_from_config) {
+		if (time_levs_from_config) {
+			fortprintf(fd, "      allocate(%s(config_flag_time_levs))\n", pointer_name);
+			fortprintf(fd, "      do iTime = 1,config_flag_time_levs\n");
```

Comment:
Here I move the loop over time levels to the auto-generated Fortran code. This one is used if a `namelist:config_name` is specified.

Reply:
Note, the flag in your `Registry.xml` file can be named whatever you want. This variable name, `config_flag_time_levs`, is just a pointer to the flag that you specify with the `namelist:` prefix.